src/app/components/tree/tree.component.ts

```diff
@@ -50,9 +54,15 @@ export class TreeComponent {
  */
   nodeSelect(_event?: any) {
     if (_event){
-        this.clickEvent.emit(_event);
+      if (this.lastClickedNodeName == null || this.lastClickedNodeName != _event.node.name) {
+      this.lastClickedNodeName = _event.node.name;
+      this.clickEvent.emit(_event); 
+      setTimeout( () => (this.lastClickedNodeName = null), 500); // < 500 ms becomes a double click
```

Comment:
500ms should not be hardcoded here. But, it is OK for it to be a constant or variable with a default value, found at the top of the file. Because, in the future we will want to know the reason, and be able to customize it perhaps.